### PR TITLE
[44 ,45] Leave a review (as a Buyer and as a Seller)

### PR DESCRIPTION
**Pull request by SaidganiUsm**
Where
[NL-45 Leave a review (as a Buyer)](https://jjbcapital.atlassian.net/browse/NL-45)
[NL-44 Leave a review (as a Seller)](https://jjbcapital.atlassian.net/browse/NL-44)

What
Users can rate a seller via purchased by him lot

How
Developed new logic for saving rates in db and validating the request (are user is a buyer of the current lot or not) 

Affected Area
Auctionify.API.ClientApp
Auctionify.Application.Features
Auctionify.Core
Auctionify.Infrastructure
Auctionify.UnitTests

PR Checklist
Build

 Build Succeeded
Unit Tests

[x] All 89 Tests Passed, 0 Ignored, 0 Failed

### CODE REVIEW

**Review by otash8110 — APPROVED** [review]

**Review by khasanrashidov — DISMISSED** [review]

**Review by khasanrashidov — DISMISSED** [review]

File: src/Auctionify.API/ClientApp/src/app/components/general/lot-profile/lot-profile.component.html [hunk 1]
```diff
@@ -328,5 +328,37 @@ <h3 *ngIf="!bidsToShow || bidsToShow.length <= 0">No bids yet!</h3>
                 </ul>
             </div>
         </div>
+        <div
+            class="right__rate-btn"
+            *ngIf="
+                currentUserId == lotData.seller?.id && lotData.lotStatus.id == 6
```
Comment (Muhammad-gf) [inline src/Auctionify.API/ClientApp/src/app/components/general/lot-profile/lot-profile.component.html:334]: What does the number '6' mean? I would suggest turning it into a variable with meaningful naming.

File: src/Auctionify.API/ClientApp/src/app/components/general/rate-user/rate-user.component.html [hunk 2]
```diff
@@ -0,0 +1,77 @@
+<div class="profile">
+    <div class="profile__top-container">
+        <ng-container *ngIf="isUserSeller(); else buyerTemplate">
```
Comment (Muhammad-gf) [inline src/Auctionify.API/ClientApp/src/app/components/general/rate-user/rate-user.component.html:3]: I would suggest moving these ngif statements to the component.ts file. Making a specific variable for the img src and setting the path after checking isUserSeller().

File: src/Auctionify.API/ClientApp/src/app/components/general/rate-user/rate-user.component.html [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            <img [src]="lotData?.seller?.profilePicture || '../../../../../assets/images/User.png'" alt="Seller-profile-image" width="100" height="100" />
+        </ng-container>
+        <ng-template #buyerTemplate>
+            <img [src]="lotData?.buyer?.profilePicture || '../../../../../assets/images/User.png'" alt="Buyer-profile-image" width="100" height="100" />
+        </ng-template>
+        <img class="profile__lot-main-photo"
+            [src]="
+                lotData && lotData.photosUrl?.[0]
```
Comment (Muhammad-gf) [inline src/Auctionify.API/ClientApp/src/app/components/general/rate-user/rate-user.component.html:11]: I think you can use the above approach also here.

File: src/Auctionify.API/ClientApp/src/app/components/general/rate-user/rate-user.component.ts [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        const rateData: RateUserCommandModel = {
+            lotId: this.lotId,
+            comment: this.comment,
+            ratingValue: this.selectedRating,
+        };
+
+        this.isLoading = true;
+
+        if (this.authorizeService.isUserSeller()) {
+            this.client.addRateToBuyer(rateData).subscribe(
```
Comment (Muhammad-gf) [inline src/Auctionify.API/ClientApp/src/app/components/general/rate-user/rate-user.component.ts:68]: Please, update the subscribe body by this article - https://rxjs.dev/deprecations/subscribe-arguments

File: src/Auctionify.API/ClientApp/src/app/components/general/rate-user/rate-user.component.ts [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+                (response) => {
```
Comment (Muhammad-gf) [inline src/Auctionify.API/ClientApp/src/app/components/general/rate-user/rate-user.component.ts:69]: You can make the body of the response as a separate method and use it twice.

File: src/Auctionify.API/ClientApp/src/app/components/general/rate-user/rate-user.component.ts [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+                    this.errorMessage = '';
+                    this.showSnackBar('Rate submitted successfully', 'success');
+                    this.isLoading = false;
+                    this.router.navigate(["/home"])
+                },
+                (error) => {
```
Comment (Muhammad-gf) [inline src/Auctionify.API/ClientApp/src/app/components/general/rate-user/rate-user.component.ts:75]: The same here.

File: src/Auctionify.API/ClientApp/src/app/components/general/rate-user/rate-user.component.ts [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+            verticalPosition: 'bottom',
+            panelClass: panelClass,
+        });
+    }
+
+    onStarClick(rating: number): void {
+        this.selectedRating = rating;
+    }
+
+    getUserProfileImage(src: string | undefined): string {
```
Comment (Muhammad-gf) [inline src/Auctionify.API/ClientApp/src/app/components/general/rate-user/rate-user.component.ts:128]: I think this method is not used anywhere, isn't it? You can remove it.

**Review by Muhammad-gf — CHANGES_REQUESTED** [review]

File: src/Auctionify.API/ClientApp/src/app/components/general/rate-user/rate-user.component.ts (hunk 7 above)
Comment (SaidganiUsm) [reply src/Auctionify.API/ClientApp/src/app/components/general/rate-user/rate-user.component.ts:128]: I'm using it in HTML file

**Review by Muhammad-gf — DISMISSED** [review]

**Comment by khasanrashidov** [comment]
fix your branch, it broke after merging the dev, because of new features, the dev has some significant and important changes:
image

**Review by Muhammad-gf — APPROVED** [review]

**Review by khasanrashidov — APPROVED** [review]

**Pull request merged**